draft-ietf-rtcweb-jsep.xml

```diff
@@ -4444,6 +4447,337 @@ a=msid:-
+      which enables the audio and video m= sections, and thereby
+      bidirectional media transmission. The advantage of such a
+      flow is that as soon as the first answer is received, the implementation
+      can proceed with ICE and DTLS negotiation and establish the session
+      transport. If the transport setup completes before the second offer is
+      sent, then media can be transmitted immediately by the callee immediately
+      upon answering the call, minimizing perceived post-dial-delay.
+      The second offer/answer exchange can also change the preferred codecs
+      or other information.
+      </t>
+
+      <t><figure>
+       <artwork><![CDATA[
+//                  set up local media state
+AliceJS->AliceUA:   create new PeerConnection
+AliceJS->AliceUA:   addTrack with two tracks: audio and video
+AliceJS->AliceUA:   createOffer to get offer
+AliceJS->AliceUA:   setLocalDescription with offer
+AliceUA->AliceJS:   multiple onicecandidate events with candidates
+
+//                  wait for ICE gathering to complete
+AliceUA->AliceJS:   onicecandidate event with null candidate
+AliceJS->AliceUA:   get |offer-C1| from value of localDescription
+
+//                  |offer-C1| is sent over signaling protocol to Bob
+AliceJS->WebServer: signaling with |offer-C1|
+WebServer->BobJS:   signaling with |offer-C1|
+
+//                  |offer-C1| arrives at Bob
+BobJS->BobUA:       create a PeerConnection
+BobJS->BobUA:       setRemoteDescription with |offer-C1|
+BobUA->BobJS:       onaddtrack events for audio and video
+
+//                  Bob prepares an early answer to warm up the transport
+BobJS->BobUA:       addTransceiver with null audio and video tracks
+BobJS->BobUA:       transceiver.setDirection(sendonly) for both transceivers
+BobJS->BobUA:       createAnswer
+BobJS->BobUA:       setLocalDescription with answer
+BobUA->BobJS:       multiple onicecandidate events with candidates
+
+//                  wait for ICE gathering to complete
+BobUA->BobJS:       onicecandidate event with null candidate
+BobJS->BobUA:       get |answer-C1| from value of localDescription
+
+//                  |answer-C1| is sent over signaling protocol to Alice
+BobJS->WebServer:   signaling with |answer-C1|
+WebServer->AliceJS: signaling with |answer-C1|
+
+//                  |answer-C1| (sendonly) arrives at Alice
+AliceJS->AliceUA:   setRemoteDescription with |answer-C1|
+AliceUA->AliceJS:   onaddtrack events for audio and video
+
+//                  ICE and DTLS establish while call is ringing
+
+//                  Bob accepts call, starts media, and sends a new offer
+BobJS->BobUA:       transceiver.setTrack with audio and video tracks
+BobJS->BobUA:       transceiver.setDirection(sendrecv) for both transceivers
+BobJS->BobUA:       createOffer
+BobJS->BobUA:       setLocalDescription with offer
+BobUA->AliceUA:     media sent from Bob to Alice
+[TODO: ice restart]
```

Comment:
I thought this example might be a good example of demoing ICE transport policy changes, so I left this here as a reminder. I will move this to an issue.